pkg/cloudprovider/providers/aws/device_allocator.go

```diff
@@ -40,11 +45,27 @@ type DeviceAllocator interface {
 	// name. Only the device suffix is returned, e.g. "ba" for "/dev/xvdba".
 	// It's up to the called to add appropriate "/dev/sd" or "/dev/xvd" prefix.
 	GetNext(existingDevices ExistingDevices) (mountDevice, error)
+
+	// LockForUse will lock the device allocator, so as another goroutine can't use it
+	LockForUse()
```

Comment:
One option could be to return a Lease on the device, and we can return devices to the pool with an indication as to whether they should be returned to the front or the back of the queue.